File: app/src/main/java/com/eduminication/ui/data/dataViewModel.java
package com.eduminication.ui.data;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class dataViewModel extends ViewModel {
    private MutableLiveData<String> mText;

    public dataViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is share fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/com/eduminication/ui/data/DataViewModel.kt
package com.eduminication.ui.data

import android.content.Context
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import java.net.URL

class DataViewModel(
    val context: Context,
    val filePath: MutableLiveData<URL> = MutableLiveData(
        URL("file:///" + context.getExternalFilesDir(null) + "/test.pptx")
    )
) : ViewModel()
